### PR TITLE
W-15534157 pdk conditional UI pdk gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 1]
```diff
@@ -163,6 +163,48 @@ image:sensitive-data.png[The character inputs to the key parameter appear as bla
...
+            "@visibleOn":
+                - property: encryptBody
+                  value: "true"
+required:
+    - encryptBody
+----
+
+Conditional parameters have the following restrictions:
+
+* You can only define single condition per parameter.
```
Comment (NachoRaik) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]:
```suggestion
* You can only define a single condition per parameter.
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 2]
```diff
@@ -163,6 +163,48 @@ image:sensitive-data.png[The character inputs to the key parameter appear as bla
...
+* Required properties that are conditionally rendered must have a default value defined.
```
Comment (NachoRaik) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]: Maybe mention here that in the example, if `encryptionKey` is required, then it should have a default value.

**Review by NachoRaik — APPROVED** [review]
LGTM

File: pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 3]
```diff
@@ -163,6 +163,46 @@ image:sensitive-data.png[The character inputs to the key parameter appear as bla
 
 Toggling *Show* reveals any new data currently being entered. However, you cannot reveal previously stored data. For example, if you configure the example *Key* parameter and apply the policy, you cannot view the parameter if you edit the policy's configuration in the future.
 
+=== Define Conditional Parameters
+
+Render policy configuration parameters dynamically by establishing conditions to make fields visible. In the following code example, the `encryptionKey` parameter only appears if the user selects `true` for the `encryptBody` parameter:
```
Comment (Crispy-Salesforce) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]:
```suggestion
Render policy configuration parameters dynamically by establishing conditions to make fields visible. In the following code example, the `encryptionKey` parameter appears if the user selects `true` for the `encryptBody` parameter:
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 4]
```diff
@@ -163,6 +163,46 @@ image:sensitive-data.png[The character inputs to the key parameter appear as bla
...
+        type: string
+        "@rendering":
+            "@visibleOn":
+                - property: encryptBody
+                  value: "true"
+----
+
+Conditional parameters have the following restrictions:
+
+* You can only define a single condition per parameter.
```
Comment (Crispy-Salesforce) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc:202]:
```suggestion
* You can define a single condition per parameter.
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 5]
```diff
@@ -163,6 +163,46 @@ image:sensitive-data.png[The character inputs to the key parameter appear as bla
...
+* Required properties that are conditionally rendered must have a default value defined. In the above examples, if `encryptionKey` is required, it requires a default value.
+* Only root-level parameters support conditional rendering. Inner properties of arrays or objects do not support conditional rendering.
```
Comment (Crispy-Salesforce) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc:204]:
```suggestion
* Root-level parameters support conditional rendering. Inner properties of arrays or objects do not support conditional rendering.
```

**Review by Crispy-Salesforce — APPROVED** [review]

**Pull request merged**